src/hydro_forces.cpp

```diff
@@ -353,58 +353,48 @@ double TestHydro::setVelHistory(double val, int step, int b_num, int index) {
 *******************************************************************************/
 std::vector<double> TestHydro::ComputeForceHydrostatics() {
 	std::vector<double> position, displacement;
-	unsigned total_dofs = 6 * num_bodies;
-	position.resize(total_dofs, 0.0);
-	displacement.resize(total_dofs, 0.0);
+	assert(num_bodies > 0);
 
-	// orientation initialized to system current pos/rot vectors
 	for (int b = 0; b < num_bodies; b++) {
-		for (int i = 0; i < 3; i++) {
-			unsigned b_offset = 6 * b;
-			if (b_offset + i + 3 > total_dofs || b_offset + i < 0) {
-				std::cout << "temp index in hydrostatic force is bad " << std::endl;
-			}
-			position[i + b_offset] = bodies[b]->GetPos()[i];
-			position[i + 3 + b_offset] = bodies[b]->GetRot().Q_to_Euler123()[i]; // TODO check if we want this, or GetWvel_par()?
+		// initialize variables
+		auto body = bodies[b];
+		auto body_h5file = file_info[b];
+		auto rho = body_h5file.rho;
+		unsigned b_offset = 6 * b;  // force_hydrostatic has 6 elements for each body so to skip to the next body we move 6 spaces
+		auto body_force_hydrostatic = &force_hydrostatic[b_offset];
+		auto gg = body->GetSystem()->Get_G_acc().Length();
+
+		// hydrostatic stiffness due to offset from equilibrium
+		auto body_position = body->GetPos();
+		auto body_rotation = body->GetRot().Q_to_Euler123();
```

Comment:
I don't think auto should be used here, instead use the type name for each variable to make the code more clear/legible.

Reply:
I am a proponent of "using auto whenever possible" because it helps with code maintainability (less prone to breaking from minor changes such as renaming types or changing templates) and it is very easy to know the type from auto with modern IDEs. From Microsoft: https://learn.microsoft.com/en-us/cpp/cpp/auto-cpp

> We recommend that you use the auto keyword for most situations—unless you really want a conversion—because it provides these benefits:
> - Robustness: If the expression's type is changed—including when a function return type is changed—it just works.
> - Performance: You're guaranteed that there's no conversion.
> - Usability: You don't have to worry about type name spelling difficulties and typos.
> - Efficiency: Your coding can be more efficient.

I changed it for simple variables such as double, let me know if you also still want to change the other declarations for more complicated types such as Chrono or Eigen types.

Reply:
All variables declared explicitly for readability (see discussion below)